pkg/didcomm/common/service/service.go

```diff
@@ -122,3 +132,22 @@ func (m *DIDCommMsg) ThreadID() (string, error) {
 
 	return "", ErrThreadIDNotFound
 }
+
+// MessageHeader is header of the generic message
+type MessageHeader struct {
+	*Header
+	Purpose  string `json:"purpose"`
+	DataType string `json:"data-type"`
+}
+
+// NewMessageHeader returns MessageHeader from given payload
+func NewMessageHeader(payload []byte) (*MessageHeader, error) {
+	msg := &MessageHeader{}
+
+	err := json.Unmarshal(payload, &msg)
```

Comment:
Don't want more Unmarshals.

Reply:
It is Same as existing function in same file NewDIDCommMsg
https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/common/service/service.go#L90

Reply:
I know.

The problem is that InboundMessageHandler (https://github.com/hyperledger/aries-framework-go/pull/1033/files#diff-9988f5820bf05228e5657f2842fa071eR107) will now unmarshal twice in these cases.

Reply:
Removed it